ThirdPartyModules/Amasty/Rewards.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\ThirdPartyModules\Amasty;
+
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Discount;
+use Bolt\Boltpay\Helper\Shared\CurrencyUtils;
+
+class Rewards
+{
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnagHelper;
+    
+    /**
+     * @var Discount
+     */
+    protected $discountHelper;
+
+    /**
+     * @param Bugsnag $bugsnagHelper Bugsnag helper instance
+     * @param Discount $discountHelper
+     */
+    public function __construct(
+        Bugsnag   $bugsnagHelper,
+        Discount  $discountHelper
+    ) {
+        $this->bugsnagHelper   = $bugsnagHelper;
+        $this->discountHelper  = $discountHelper;
+    }
+
+    /**
+     * @param array $result
+     * @param Amasty\Rewards\Helper\Data $amastyRewardsHelperData
+     * @param Quote $quote
+     * 
+     * @return array
+     */
+    public function collectDiscounts($result,
+                                     $amastyRewardsHelperData,
+                                     $quote,
+                                     $paymentOnly)
```

Comment:
@dumega I added the parent quote as a parameter in for the filter collectDiscounts in this PR https://github.com/BoltApp/bolt-magento2/pull/931/files#diff-d9d65ba4532ae42767de8b35f1843726R2178
Can you re-base the branch to master and add it as well?

Reply:
fixed